File: Array/1030.Matrix_Cells_in_Distance_Order/C++/1030.Matrix_Cells_in_Distance_Order.cpp
class Solution {
public:
    vector<vector<int>> allCellsDistOrder(int rows, int cols, int rCenter, int cCenter) 
    {
        //=============================================================//
        // - Input:                                                    //
        //   int (i.e. matrix rows)                                    //
        //   int (i.e. matrix columns)                                 //
        //   int (i.e. center of matrix rows)                          //
        //   int (i.e. center of matrix columns)                       //
        // - Return:                                                   //
        //   vector<vector<int>> (i.e. coordinates in sorted distance) //
        //=============================================================//

        /*Initialize*/
        ///// Result coordinate vector /////
        vector<vector<int>> res_coordnte_vect;

        ///// Temporal dictionary/hashmap, and its length /////
        map<int, vector<vector<int>>> tmp_map;
        int tmp_map_len;

        for (int tmp_row_idx = 0; tmp_row_idx < rows; tmp_row_idx++) //Rows
        {
            for (int tmp_col_idx = 0; tmp_col_idx < cols; tmp_col_idx++) //Columns
            {
                int tmp_distance = (abs(tmp_row_idx-rCenter) + abs(tmp_col_idx-cCenter)); //Temporal coordinate distances

                ///// Check if existing distances in hashmap or not /////
                if (tmp_map.find(tmp_distance) == tmp_map.end())
                {
                    tmp_map[tmp_distance] = {{tmp_row_idx, tmp_col_idx}}; //Insertion initialize
                }
                else
                {
                    tmp_map[tmp_distance].push_back({tmp_row_idx, tmp_col_idx}); //Insertion resume
                }
            } //Columns

        } //Rows

        res_coordnte_vect = tmp_map[0]; //Query first
        tmp_map_len = (tmp_map.size()); //Get length from hashmap


        /*Greedy-based loop traversal for coordinate with sorted distances*/
        for (int tmp_map_idx = 1; tmp_map_idx < tmp_map_len; tmp_map_idx++) //Whole
        {
            res_coordnte_vect.insert(res_coordnte_vect.end(), tmp_map[tmp_map_idx].begin(), tmp_map[tmp_map_idx].end()); //Insertion resume

        } //Whole

        return res_coordnte_vect;
    }
};

File: Array/1030.Matrix_Cells_in_Distance_Order/Python/1030.Matrix_Cells_in_Distance_Order.py
class Solution(object):
    def allCellsDistOrder(self, rows, cols, rCenter, cCenter):
        """
        :type rows: int
        :type cols: int
        :type rCenter: int
        :type cCenter: int
        :rtype: List[List[int]]
        """
        #===============================#
        # Greedy-based traversal method #
        #===============================#

        ############
        #Initialize
        ##### Result array #####
        res_arry = []

        ##### Temporal dictionary/hashmap #####
        tmp_dict = {}

        for row_idx in range(rows):
            for col_idx in range(cols):
                if ((abs(rCenter-row_idx) + abs(cCenter-col_idx)) in tmp_dict):
                    tmp_dict[(abs(rCenter-row_idx) + abs(cCenter-col_idx))].append([row_idx, col_idx])
                else:
                    tmp_dict[(abs(rCenter-row_idx) + abs(cCenter-col_idx))] = [[row_idx, col_idx]]

        tmp_dict_len = len(tmp_dict)


        ###########################################################
        #Greedy-based loop traversal with coordinate distances map
        for tmp_idx in range(tmp_dict_len):
            res_arry += tmp_dict[tmp_idx] #Concate/Extend coordinate distances in sorted order

        return res_arry
